modules/core/src/lib/layer-props.ts

```diff
@@ -0,0 +1,64 @@
+import {COORDINATE_SYSTEM} from './constants';
+import type Layer from './layer';
+
+interface NonIterableData {
+  length: number;
+  attributes?: Record<string, BinaryAttribute>;
+}
+type LayerData = Iterable<any> | NonIterableData;
+
+// TODO
+type PickingInfo = any;
+type MjolnirEvent = any;
+type BinaryAttribute = any;
+
+export interface LayerProps {
```

Comment:
Putting a JSDoc string in front of each type is nice, it helps when hovering in vscode.